spec.emu

```diff
@@ -12,162 +12,113 @@ copyright: false
-        <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-inclusive-getter">
-            <h1>get %NumericRangeIteratorPrototype%.isEnclusiveEnd</h1>
-            <emu-alg>
-                1. Perform ? RequireInternalSlot(*this*, [[inclusiveEnd]]).
-                1. Return *this*.[[inclusiveEnd]].
-            </emu-alg>
+    <emu-clause id="sec-iteration" number="1">
+        <h1>Iteration</h1>
+
+        <emu-clause id="sec-properties-of-the-iterator-constructor" number="2">
+            <h1>Properties of the Iterator Constructor</h1>
+
+            <emu-clause id="sec-iterator.range">
+                <h1>Iterator.range ( _start_, _end_, _option_ )</h1>
+                <emu-alg>
+                    1. If _start_ is a Number, return ? CreateNumericRangeIterator(_start_, _end_, _option_, ~number-range~).
+                    1. If _start_ is a BigInt, return ? CreateNumericRangeIterator(_start_, _end_, _option_, ~bigint-range~).
+                    1. Throw a *TypeError* exception.
+                </emu-alg>
+            </emu-clause>
         </emu-clause>
-    </emu-clause>
-    <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-step">
-        <h1>%NumericRangeIteratorPrototype%.step</h1>
-        <p>%NumericRangeIteratorPrototype%.step is an accessor property with attributes { [[Enumerable]]: false, [[Configurable]]: *true*}. The [[Get]] attribute is defined as follows:</p>
-        <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-step-getter">
-            <h1>get %NumericRangeIteratorPrototype%.step</h1>
-            <emu-alg>
-                1. Perform ? RequireInternalSlot(*this*, [[step]]).
-                1. Return *this*.[[step]].
-            </emu-alg>
+
+        <emu-clause id="sec-numeric-range-iterator-object">
+            <h1>The <dfn>NumericRangeIterator</dfn> Object</h1>
+            <p>A NumericRangeIterator object is an iterator that yields numbers. There is not a named constructor for NumericRangeIterator objects. Instead, NumericRangeIterator objects are created by the CreateNumericRangeIterator abstract operation as needed.</p>
+
+            <emu-clause id="sec-create-numeric-range-iterator" type="abstract operation">
+                <h1>
+                    CreateNumericRangeIterator (
+                        _start_: a Number or a BigInt,
+                        _end_: an ECMAScript language value,
+                        _option_: an ECMAScript language value,
+                        _type_: ~number-range~ or ~bigint-range~
+                    )
+                </h1>
+                <dl class="header"></dl>
+                <emu-alg>
+                    1. If _type_ is ~number-range~, let _primitiveType_ be *"number"*, else let _primitiveType_ be *"bigint"*.
+                    1. Assert: Type(_start_) is _primitiveType_.
```

Comment:
I’m not sure this works - Type returns Number and BigInt. You may want an if/else with an assert (and the zero/ones) in both branches.